### PR TITLE
Added branch names generation and exported magic numbers to consts.

### PR DESCRIPTION
**Pull request by sandspider2234**

### CODE REVIEW

File: .management/branch_names.py [hunk 1]
```diff
@@ -1,30 +1,85 @@
...
+        i += 1
+    return set(names_set)
+
+
+def create_and_push_branch(repo, branch_name):
+    """
+    Creates a branch with the given name,
+    :param repo: Git repo object.
+    :param branch_name: Name of branch to create.
+    :return:
```
Comment (ShayNehmad) [inline .management/branch_names.py]: If doesn't return anything, delete this

File: .management/branch_names.py [hunk 2]
```diff
@@ -1,30 +1,85 @@
...
+        if random_name in names_set:
+            continue
+        names_set.append(random_name)
+        i += 1
+    return set(names_set)
+
+
+def create_and_push_branch(repo, branch_name):
+    """
+    Creates a branch with the given name,
```
Comment (ShayNehmad) [inline .management/branch_names.py]: The suspense is killing me

File: .management/branch_names.py [hunk 3]
```diff
@@ -1,30 +1,85 @@
...
+    :param repo: Git repo object.
+    :param branch_name: Name of branch to create.
+    :return:
+    """
+
+    git_cmd = repo.git
+    git_cmd.checkout('HEAD', b=branch_name)
+    file_path = os.path.join(repo.working_dir, "{0}.txt".format(branch_name))
```
Comment (ShayNehmad) [inline .management/branch_names.py]: Can we put this under `.management/`, just so it doesn't pollute the root dir?  (and if so put it in a param)

File: .management/branch_names.py [hunk 4]
```diff
@@ -1,30 +1,85 @@
...
+
+    with open(file_path, 'w') as fd:
+        fd.write(branch_name)
+
+    repo.index.add(file_path)
+    repo.index.commit(branch_name)
```
Comment (ShayNehmad) [inline .management/branch_names.py:90]: This way you can filter all these commits out by looking at the ones who have messages only. Can we put the branch name as a message/some other random words?

File: .management/branch_names.py [hunk 5]
```diff
@@ -1,30 +1,85 @@
...
 
 def main():
     print("I'm the branch name generator.\nHere are some names for you:\n")
-    # Think about changing this to movie names from imdb database
     words = load_words()
     print(len(words))
 
     # Filter to mostly normal pronounceable words
-    words = set(filter(lambda x: 2 < len(x) < 9, words))
+    words = set(filter(lambda x: MINIMUM_WORDS_LENGTH < len(x) < MAXIMUM_WORDS_LENGTH, words))
```
Comment (ShayNehmad) [inline .management/branch_names.py]: This range ought to be params (with argparse)

File: .management/branch_names.py [hunk 6]
```diff
@@ -1,30 +1,85 @@
...
 
     print(len(words))
+    names_set = generate_names_set(words)
+    print(names_set)
 
-    while True:
-        random_name = '-'.join(random.sample(words, 3))
-        print(random_name)
-        _ = input()
+    repo_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```
Comment (ShayNehmad) [inline .management/branch_names.py]: This should either be a command line argument, or change to use Pathlib

**Review by ShayNehmad — CHANGES_REQUESTED** [review]

File: .management/branch_names.py (hunk 1 above)
Comment (sandspider2234) [reply .management/branch_names.py]: Deleted.

File: .management/branch_names.py (hunk 3 above)
Comment (sandspider2234) [reply .management/branch_names.py]: Done

File: .management/branch_names.py (hunk 5 above)
Comment (sandspider2234) [reply .management/branch_names.py]: Done

File: .management/branch_names.py (hunk 6 above)
Comment (sandspider2234) [reply .management/branch_names.py]: Done

**Pull request merged**